### DISCUSSION TITLE
[Feature Request] To save metadata into google firestore database

### DISCUSSION THREAD

**Discussion by diyism** [juicedata/juicefs#1963] [Ideas]
Currently I'm using juicefs in google colab VM like this:

```
from google.colab import drive
drive.mount('/root/gdrive')
%cd /root/

!install /root/gdrive/MyDrive/ColabSingularity/juicefs /usr/local/bin

#!juicefs format --storage file --bucket /root/gdrive/MyDrive/ColabSingularity/gdrive_file_volume1 "sqlite3:///root/gdrive_file_volume1.db" volume1
!while true; do nohup cp /root/gdrive_file_volume1.db /root/gdrive/MyDrive/ColabSingularity/ > /dev/null 2>&1; sleep 10; done &
!juicefs mount -d "sqlite3:///root/gdrive/MyDrive/ColabSingularity/gdrive_file_volume1.db" ./gdrive_file_volume1
```

the object data successfully saved into my google drive in real time,
but the metadata only can be saved into my google drive every 10 seconds.

Since the google colab VM will be shutdown every 12 hours, there's a chance that  my metadata sqlite file may be corrupted.
So could you support the google firestore database as juicefs metadata engine?

![Screenshot 2022-05-04 at 23-01-25 diyism – Cloud Firestore – Firebase console](https://user-images.githubusercontent.com/1744399/166710581-3c720341-8e88-4194-afc1-23abe286f40d.png)

**Comment by davies** [comment]
@diyism Can you use managed SQL database other than sqlite?

Google firestore may not fit well as metadata engine, we don't have a plan to support it.
